src/net_processing.cpp

```diff
@@ -4525,12 +4741,12 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                     while (!vInvTx.empty() && nRelayedTransactions < INVENTORY_BROADCAST_MAX) {
                         // Fetch the top element from the heap
                         std::pop_heap(vInvTx.begin(), vInvTx.end(), compareInvMempoolOrder);
-                        std::set<uint256>::iterator it = vInvTx.back();
+                        uint256 hash = vInvTx.back();
+                        const bool flood_tx = pto->m_tx_relay->m_txs_to_announce.at(hash);
```

Comment:
IIUC, erlay is introducing an interdependency between transaction's source and selection of outgoing announcement-protocol (cf `RelayTransaction`), but I'm not if this is covered by `p2p_erlay.py` ? Following diff doesn't yell a failure. Same we turning to always true.

```
diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index 1818fc0f2..a4e847110 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -4732,7 +4732,8 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                         // Fetch the top element from the heap
                         std::pop_heap(vInvTx.begin(), vInvTx.end(), compareInvMempoolOrder);
                         uint256 hash = vInvTx.back();
-                        bool flood_tx = pto->m_tx_relay->m_txs_to_announce.at(hash);
+                        //bool flood_tx = pto->m_tx_relay->m_txs_to_announce.at(hash);
+                        bool flood_tx = false;
                         vInvTx.pop_back();
                         CInv inv(state.m_wtxid_relay ? MSG_WTX : MSG_TX, hash);
                         // Remove it from the to-be-sent set
```